hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllOperation.java

```diff
@@ -145,8 +148,16 @@ private Object putToRecordStore(Data dataKey, Data dataValue) {
 
     @Override
     protected void afterRunInternal() {
-        invalidateNearCache(invalidationKeys);
+        // Publish listener events
+        // This should happen in after-run to avoid surfacing exceptions during predicate attribute conversion.
+        // Single Map.put() behaves the same. see. https://github.com/hazelcast/hazelcast/issues/15415
+        for (EventDescriptor descriptor : unpublishedEvents) {
+            mapEventPublisher.publishEvent(descriptor.caller, descriptor.mapName,
```

Comment:
Can we catch and log exceptions inside `publishEvent` method of `MapEventPublisherImpl`?

Reasonings:

1. Centralize place for all map operations, since some other map operations have still same issue.
2. We can catch exceptions per listener basis to minimize blast of possible failure. For instance, we have attached 100 listener to map but only one of them is throwing exceptions, this failure shouldn't block event flow to other listener.

Reply:
@ahmetmircik followed your recommendation. It seemed a bit more risky, but I agree, the API is not perfect to handle errors, when it comes to bulk ops like this, one failure shouldn't block the rest.